install/tools/man/ipa-migrate.1

```diff
@@ -0,0 +1,111 @@
+The host name of the remote IPA server that is being migrated from.
+.TP
+\fB\-D\fR, \fB\-\-bind\-dn=BIND_DN\fR
+The Bind DN (Distinguished Name) or an LDAP entry to bind to the remote IPA server with.
+Typically this is "cn=directory manager", but it could be any entry that has
+access to read the userPassword attribute.
+.TP
+\fB\-w\fR, \fB\-\-bind\-pw=PASSWORD\fR
+The password for the Bind DN that is authenticating against the remote IPA server.  If
+a password is not provided then the tool with prompt for the password if needed.
+.TP
+\fB\-Just\fR, \fB\-\-bind\-pw\-file=FILE_PATH\fR
+Path to a file containing the password for the Bind DN.
+.TP
+\fB\-Z\fR, \fB\-\-cacertfile=FILE_PATH\fR
+Path to a file containing a CA Certificate that the remote server trusts
+.TP
+\fB\-l\fR, \fB\-\-log\-file=FILE_PATH\fR
+Path to a file containing the migration log.  By default the tool will use \fI/var/log/ipa-migrate.log\fR
+.TP
+\fB\-x\fR, \fB\-\-dryrun\fR
+Go through the migration process but do not write and data to the new IPA server.
+.TP
+\fB\-o\fR, \fB\-\-dryrun\-record=FILE_PATH\fR
+Go through the migration process but do not write any data to the new IPA server. However, write the
+migration operations to an LDIF file which can be applied later or reused for multiple migrations.
+.TP
+\fB\-r\fR, \fB\-\-reset\-range\fR
+Reset the ID range for migrated users/groups. In "stage-mode" this is done automatically
+.TP
+\fB\-F\fR, \fB\-\-force\fR
+Ignore any errors and continue to proceed with migration effort.
+.TP
+\fB\-V\fR, \fB\-\-version\fR
+Display the version of the migration tool.
+.TP
+\fB\-q\fR, \fB\-\-quiet\fR
+Only log errors during the migration process.
+.TP
+\fB\-B\fR, \fB\-\-skip\-dns\fR
+Do not migrate DNS records
+.TP
+\fB\-S\fR, \fB\-\-skip\-schema\fR
+Do not migrate the database schema
+.TP
+\fB\-C\fR, \fB\-\-skip\-config\fR
+Do not migrate the database configuration (dse.ldif/cn=config)
+.TP
+\fB\-O\fR, \fB\-\-schema\-overwrite\fR
+Overwrite existing schema definitions.  By default duplicate schema is skipped.
+.TP
+\fB\-s\fR, \fB\-\-subtree=DN\fR
+Specifies a custom database subtree that should be included in the migration.
+This is only needed if non-default subtrees/branches were added to the database
+outside of IDM.
+.TP
+\fB\-f\fR, \fB\-\-db\-ldif=FILE_PATH\fR
+LDIF file containing the entire backend. If omitted the tool will query the remote IPA server.
+.TP
+\fB\-m\fR, \fB\-\-schema\-ldif=FILE_PATH\fR
+LDIF file containing the schema. If omitted the tool will query the remote IPA server.
+.TP
+\fB\-g\fR, \fB\-\-config\-ldif=FILE_PATH\fR
+LDIF file containing the entire "cn=config" DIT. If omitted the tool will query the remote IPA server.
+.TP
+\fB\-n\fR, \fB\-\-no\-prompt\fR
+Do not prompt for confirmation before starting migration.  Use at your own risk!
+.TP
+
+.SH "EXIT STATUS"
+0 if the command was successful
+
+1 if an error occurred
+
+2 if the host is not a IPA server, or the IPA server installation is faulty
```

Comment:
I see three conditions where 2 would be returned (but no check for whether the remote hostname is an IPA server
* realm cannot be identified, "Unable to find realm from remote LDIF"
* IPA is not configured
* the API can't be initialized, "Problem with IPA installation"

Reply:
To check whether the remote is an IPA server do a base search on the suffix naming context. The attribute info will contain "IPA V2.0".  We never saw the need to bump it further.